backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Text.Json;
+using Api.MQTT.Events;
+using Api.MQTT.MessageModels;
+using MQTTnet;
+using MQTTnet.Client.Connecting;
+using MQTTnet.Client.Disconnecting;
+using MQTTnet.Client.Options;
+using MQTTnet.Client.Receiving;
+using MQTTnet.Extensions.ManagedClient;
+
+namespace Api.MQTT
+{
+    public class MqttService : BackgroundService
+    {
+        #region Events
+        public event EventHandler<MqttReceivedArgs>? MqttIsarMissionReceived;
+        public event EventHandler<MqttReceivedArgs>? MqttIsarTaskReceived;
+        public event EventHandler<MqttReceivedArgs>? MqttIsarStepReceived;
+        #endregion
+
+        public static MqttService? Instance { get; private set; }
+
+        private readonly ILogger<MqttService> _logger;
+
+        private readonly IManagedMqttClient _mqttClient;
+
+        private readonly ManagedMqttClientOptions _options;
+
+        private readonly TimeSpan _reconnectDelay = TimeSpan.FromSeconds(5);
+
+        public MqttService(ILogger<MqttService> logger, IConfiguration config)
+        {
+            Instance = this;
+            _logger = logger;
+            var mqttFactory = new MqttFactory();
+            _mqttClient = mqttFactory.CreateManagedMqttClient();
+
+            var mqttConfig = config.GetSection("Mqtt");
+
+            string username = mqttConfig.GetValue<string>("Username");
+            string password = "default";
```

Comment:
Should this be collected from the environment in some manner?

Reply:
Definetily! Waiting for the keyvault